nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,362 @@
+    }
+
+    public KustoIngestionResult ingestData(KustoIngestionRequest kustoIngestionRequest) throws URISyntaxException {
+        StreamSourceInfo info = new StreamSourceInfo(kustoIngestionRequest.getInputStream());
+        //ingest data
+        IngestionResult ingestionResult;
+        IngestionProperties ingestionProperties = new IngestionProperties(kustoIngestionRequest.getDatabaseName(),
+                kustoIngestionRequest.getTableName());
+
+        IngestionMapping.IngestionMappingKind ingestionMappingKind = setDataFormatAndMapping(kustoIngestionRequest.getDataFormat(), ingestionProperties);
+        if (StringUtils.isNotEmpty(kustoIngestionRequest.getMappingName()) && ingestionMappingKind != null) {
+            ingestionProperties.setIngestionMapping(kustoIngestionRequest.getMappingName(), ingestionMappingKind);
+        }
+
+        ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FAILURES_AND_SUCCESSES);
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE);
+        ingestionProperties.setFlushImmediately(false);
+        ingestionProperties.setIgnoreFirstRecord(StringUtils.equalsIgnoreCase(kustoIngestionRequest.getIgnoreFirstRecord(), "YES"));
+
+        try {
+            if (kustoIngestionRequest.isStreamingEnabled()) {
+                ingestionResult = managedStreamingIngestClient.ingestFromStream(info, ingestionProperties);
+            } else {
+                ingestionResult = queuedIngestClient.ingestFromStream(info, ingestionProperties);
+            }
+            if (kustoIngestionRequest.pollOnIngestionStatus()) {
+                List<IngestionStatus> statuses;
+                long startTime = System.currentTimeMillis();
+                long timeoutMillis = Long.parseLong(kustoIngestionRequest.getIngestionStatusPollingTimeout()) * 1000L;
+                while (true) {
+                    // Get the status of the ingestion operation
+                    List<IngestionStatus> statuses1 = ingestionResult.getIngestionStatusCollection();
+                    if (statuses1.get(0).status == OperationStatus.Succeeded
+                            || statuses1.get(0).status == OperationStatus.Failed
+                            || statuses1.get(0).status == OperationStatus.PartiallySucceeded) {
+                        statuses = statuses1;
+                        break;
+                    }
+                    // Check if the timeout has been exceeded
+                    if (System.currentTimeMillis() - startTime >= timeoutMillis) {
+                        throw new ProcessException("Timeout exceeded while waiting for ingestion status");
+                    }
+
+                    // Sleep for 5 seconds before checking again
+                    Thread.sleep(Integer.parseInt(kustoIngestionRequest.getIngestionStatusPollingInterval()) * 1000L);
```

Comment:
See the related note on parsing this value in the calling method and using `java.time.Duration`.